deltachat-ios/DC/Wrapper.swift

```diff
@@ -18,6 +19,10 @@ class DcContext {
         dc_context_unref(contextPointer)
     }
 
+    static var shared: DcContext {
+        return .dcContext
```

Comment:
i would suggest to get rid of this, it seems to be a "better hidden" global, but will still has its disadvantages, eg. it is unclear about the timing when it is created, there cannot be two instances of DcContext etc.

also, code-wise, i think it does not make things easier, but adds some complexity.

i would suggest to just create a DcContext object in AppDelegate as in the past and pass around the dcContext pointer - in most cases, this is already done this way.

Reply:
Depends on what we want. If we want to use DcContext as a singleton,  this is the way to go. There won't be more than one instances (because the initialiser is `private`). 

We still could instantiate and set the dcObject in AppDelegate, so it will be there at the very app start.
I would generally like to avoid the pattern:
`(UIApplication.shared.delegate as! AppDelegate).dcContext`

Reply:
i think, we do not want to use DcContext as a singleton, i think, we should take the chance to use it as a normal object, this makes usage more flexible.

a "singleton" still is a global pointer under a different name, and this "shared" thingie adds some complexity and hidden things.

eg. it is completely unclear _when_ the object is actually created - in fact, even before AppDelegate is created and it is a bit luck which other objects (Bundle, ...) are available at that point - and even more bad, this may even change. i would not depend on that - also as there is no real reason for that, even if we keep a global pointer.

Reply:
Ok.
We could instantiate it in `AppDelegate.didFinishLaunchingWithOptions`.

Reply:
instancing it at the current position is okay, it is a member of AppDelegate and AppDelegate is create at some well defined point.

however, using a global value for the shared thing makes it unclear, when the shared pointer is really initialized - it may be at the moment where it is used at AppDelegate - but it may als be much sooner. even if this is documented somewhere, i think this is some complexity we should avoid. also, these kind of things are obviously hard to understand and to communicate.

that's my point.

getting rid of the global pointer will solve this issue - with the additional advantage that the use of DcContext is more flexible.

Reply:
for convenience, we may come to the point where we want to save the dcContext somewhere globally, however, this should be done outside DcContext somewhere.